rs/tests/src/orchestrator/subnet_recovery_app_subnet.rs

```diff
@@ -521,15 +580,116 @@ fn halt_subnet(
     .expect("Failed to detect broken subnet.");
 }
 
+// Corrupt the latest cup of all subnet nodes by change the CUP's replica version field.
+// This will change the hash of the block, thus making the CUP non-deserializable.
+fn corrupt_latest_cup(subnet: &SubnetSnapshot, recovery: &Recovery, logger: &Logger) {
+    const CUP_PATH: &str = "/var/lib/ic/data/cups/cup.types.v1.CatchUpPackage.pb";
+    const NEW_CUP_PATH: &str = "/var/lib/ic/data/cups/new_cup.pb";
+
+    let app_node = subnet.nodes().next().unwrap();
+    let session = app_node.block_on_ssh_session().unwrap();
+
+    info!(
+        logger,
+        "Setting journal cursor on node {:?}",
+        app_node.get_ip_addr()
+    );
+    let message_str = execute_bash_command(
+        &session,
+        "journalctl -n1 -o json --output-fields='__CURSOR'".to_string(),
+    )
+    .expect("journal message");
+    let message: Cursor = serde_json::from_str(&message_str).expect("JSON journal message");
+
+    info!(logger, "Reading CUP from node {:?}", app_node.get_ip_addr());
+    let (mut channel, _) = session.scp_recv(Path::new(CUP_PATH)).unwrap();
+    let mut bytes = Vec::new();
+    channel.read_to_end(&mut bytes).unwrap();
+
+    info!(logger, "Modifying CUP replica version");
+    let proto_cup = pb::CatchUpPackage::decode(bytes.as_slice()).unwrap();
+    let mut cup = CatchUpPackage::try_from(&proto_cup).unwrap();
+    cup.content.block.as_mut().version = ReplicaVersion::try_from("invalid_version").unwrap();
```

Comment:
Why does this make the CUP unserializable?

Reply:
See the comment above the function, whenever we deserialize a block, we also check its integrity against the block hash. If the integrity check fails we return a deserialization error